### PR TITLE
Sdk refactoring legacy

### PR DESCRIPTION
**Pull request by Velenir**
More misc + legacy stuff for SDK refactor

### CODE REVIEW

File: src/approve.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ export const constructApproveToken = (
     tokenAddress,
     overrides = {}
   ) => {
+    // @TODO consider caching `spender`
```
Comment (mwamedacen) [inline src/approve.ts:34]: good idea, gotta do it per network

Comment (Velenir) [reply src/approve.ts:34]: if cached inside `construct*` closure, no need to care about network

File: package.json [hunk 2]
```diff
@@ -47,18 +47,19 @@
   "devDependencies": {
     "@size-limit/preset-small-lib": "^7.0.8",
     "@tsconfig/recommended": "^1.0.1",
-    "axios": "^0.26.0",
+    "axios": "^0.26.1",
```
Comment (sakulstra) [inline package.json:50]: would you consider also dropping axios in favor of so like unfetch?
I personally prefer working with axios as well, but it's quite heavy on the bundle when not needed for anything else.

https://bundlephobia.com/package/axios@0.26.1
https://bundlephobia.com/package/isomorphic-unfetch@3.1.0

File: src/legacy/index.ts [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+}
+
+/** @deprecated */
+export class ParaSwap {
+  sdk: Partial<AllSDKMethods> = {};
+  fetcher: FetcherFunction;
+
+  constructor(
+    private network: number = 1,
+    private apiURL: string = API_URL,
```
[the remark this answers is not in the document]
Comment (Velenir) [reply src/legacy/index.ts:58]: [b293229](https://github.com/paraswap/paraswap-sdk/pull/111/commits/b293229fb11f663df9dfc70f1ec1ffb5274966a6)

File: package.json (hunk 2 above)
Comment (mwamedacen) [reply package.json:51]: Those are dev dependancies only. I let you discover the new SDK through the readme: https://github.com/paraswap/paraswap-sdk/blob/sdk-refactoring/README.md. Contributions are welcome for an `unfetch` wrapper.

Comment (Velenir) [reply package.json:51]: I would expect `unfetch` to just work with `fetch` wrapper, even if types may not quite match

**Pull request merged**